modules/search-service-impl/src/main/java/org/opencastproject/search/impl/SearchServiceImpl.java

```diff
@@ -443,6 +456,55 @@ public boolean deleteSynchronously(String mediaPackageId) throws SearchException
     }
   }
 
+  /**
+   * Immediately removes the given series from the search service.
+   *
+   * @param seriesId
+   *          the series
+   * @return <code>true</code> if the series was deleted
+   * @throws SearchException
+   *           if deletion failed
+   * @throws UnauthorizedException
+   *           if the user did not have access to the series
+   * @throws NotFoundException
+   *           if the series did not exist
+   */
+  public boolean deleteSeriesSynchronously(String seriesId) throws SearchException, UnauthorizedException,
+          NotFoundException {
+    SearchResult result;
+    try {
+      SearchQuery searchQuery = new SearchQuery();
+      searchQuery.withId(seriesId);
+      searchQuery.includeSeries(true);
+
+      result = solrRequester.getForWrite(searchQuery);
+      if (result.getItems().length == 0) {
+        logger.warn(
+                "Can not delete series {}, which is not available for the current user to delete from the search index.",
+                seriesId);
+        return false;
+      }
+      logger.info("Removing series {} from search index", seriesId);
+
+      Date now = new Date();
+      try {
+        persistence.deleteSeries(seriesId, now);
+        logger.info("Removed series {} from search persistence", seriesId);
+      } catch (NotFoundException e) {
+        logger.info("Could not find series with id {} in persistence, but will try remove it from index, anyway.",
+                seriesId);
+      } catch (SearchServiceDatabaseException e) {
+        logger.error("Could not delete series with id {} from persistence storage", seriesId);
+        throw new SearchException(e);
+      }
+
+      return indexManager.delete(seriesId, now);
```

Comment:
While you did delete events before in the database, you only delete the series in the index, potentially leaving database and index out of sync. The index does have a series entry btb.